encode.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+
+	vegeta "github.com/tsenart/vegeta/lib"
+)
+
+const (
+	encodingCSV  = "csv"
+	encodingGob  = "gob"
+	encodingJSON = "json"
+)
+
+type decoderFunc func(io.Reader) vegeta.Decoder
+type encoderFunc func(io.Writer) vegeta.Encoder
+
+func encodeCmd() command {
+	var (
+		fs     = flag.NewFlagSet("encode", flag.ExitOnError)
+		from   = fs.String("from", "gob", "Input decoding [csv, gob, json]")
+		to     = fs.String("to", "json", "Output encoding [csv, gob, json]")
+		output = fs.String("output", "", "Output file")
+	)
+
+	fs.Usage = func() {
+		fmt.Println("Usage: vegeta encode [flags] [<file>...]")
+		fs.PrintDefaults()
+	}
+
+	return command{
+		fs,
+		func(args []string) error {
+			fs.Parse(args)
+			return encode(*from, *to, *output, args...)
+		},
+	}
+}
+
+func encode(from, to, output string, inputs ...string) error {
+	var decFn decoderFunc
+
+	switch from {
+	case encodingCSV:
+		decFn = vegeta.NewCSVDecoder
+	case encodingJSON:
+		decFn = vegeta.NewJSONDecoder
+	default:
```

Comment:
Do we need to rely on the default case to play nice with the pipes as per your comment? The default "gob" could come from the flags outside this function.